common-test-lib/src/main/kotlin/com/google/container/tools/test/ContainerToolsRule.kt

```diff
@@ -64,33 +72,80 @@ class ContainerToolsRule(private val testInstance: Any) : TestRule {
         }
     }
 
-    private fun setUpRule() {
+    private fun setUpRule(description: Description) {
         ideaProjectTestFixture =
             IdeaTestFixtureFactory.getFixtureFactory().createLightFixtureBuilder().fixture
         EdtTestUtil.runInEdtAndWait(ThrowableRunnable { ideaProjectTestFixture.setUp() })
 
         MockKAnnotations.init(testInstance, relaxed = true)
         replaceServices()
+        createTestFiles(description.methodName)
     }
 
     private fun tearDownRule() {
+        filesToDelete.forEach { it.delete() }
         EdtTestUtil.runInEdtAndWait(ThrowableRunnable { ideaProjectTestFixture.tearDown() })
     }
 
     /**
      * Replaces all services annotated with [TestService].
      */
     private fun replaceServices() {
-        for (member in testInstance::class.declaredMemberProperties) {
-            if (member.annotations.filter { it is TestService }.isNotEmpty()) {
-                member as KProperty1<Any?, Any?>
-                member.isAccessible = true
-                val service: Any = member.get(testInstance)!!
-                setService(service)
+        for (member in getMembersWithAnnotation(TestService::class.java)) {
```

Comment:
Is there a reason why we need to use Java class here instead of native Kotlin class?